### ISSUE TITLE
Can't build Android app

### ISSUE THREAD

**Issue by ykaito21** [simolus3/moor#363]
I cannot build with this error. Is there any possible solution?

**Error:**
```
Launching lib/main.dart on AOSP on IA Emulator in debug mode...

Abi 'MIPS' is not recognized in '/Users/kaii/Library/Android/sdk/ndk/21.0.6224779'.
Abi 'MIPS64' is not recognized in '/Users/kaii/Library/Android/sdk/ndk/21.0.6224779'.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':moor_ffi:generateJsonModelDebug'.
> Build command failed.
  Error while executing process /Users/kaii/Library/Android/sdk/cmake/3.6.4111459/bin/cmake with arguments {-H/Users/kaii/development/flutter/.pub-cache/hosted/pub.dartlang.org/moor_ffi-0.3.2/android/cpp -B/Users/kaii/development/flutter/.pub-cache/hosted/pub.dartlang.org/moor_ffi-0.3.2/android/.cxx/cmake/debug/armeabi -DANDROID_ABI=armeabi -DANDROID_PLATFORM=android-22 -DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/Users/kaii/project/flutter/whatto/build/moor_ffi/intermediates/cmake/debug/obj/armeabi -DCMAKE_BUILD_TYPE=Debug -DANDROID_NDK=/Users/kaii/Library/Android/sdk/ndk/21.0.6113669 -DCMAKE_TOOLCHAIN_FILE=/Users/kaii/Library/Android/sdk/ndk/21.0.6113669/build/cmake/android.toolchain.cmake -DCMAKE_MAKE_PROGRAM=/Users/kaii/Library/Android/sdk/cmake/3.6.4111459/bin/ninja -GAndroid Gradle - Ninja}
  -- Configuring incomplete, errors occurred!

  CMake Error at /Users/kaii/Library/Android/sdk/cmake/3.6.4111459/share/cmake-3.6/Modules/CMakeDetermineSystem.cmake:104 (message):
    Could not find toolchain file:
    /Users/kaii/Library/Android/sdk/ndk/21.0.6113669/build/cmake/android.toolchain.cmake
  Call Stack (most recent call first):
    CMakeLists.txt:2 (project)
  CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
Gradle task assembleDebug failed with exit code 1
Exited (sigterm)

```

**flutter doctor:**
```
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.14.6 18G95, locale ja-JP)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
[✓] Android Studio (version 3.5)
[✓] VS Code (version 1.41.1)
[✓] Connected device (2 available)
```

**packages:**
```
  moor: ^2.3.0
  moor_ffi: ^0.3.2
  path_provider: ^1.5.1
  path: ^1.6.4

dev_dependencies:
  moor_generator: ^2.3.1
  build_runner: ^1.7.3
```

**Comment by simolus3** [comment]
Can you try adding the following to your pubspec, run `flutter clean` and then try again?
```yaml
dependency_overrides:
  moor_ffi:
    git:
      url: https://github.com/simolus3/moor.git
      ref: ndk-build
      path: moor_ffi
```

**Comment by ykaito21** [comment]
I tried it, but got this
```
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':moor_ffi:generateJsonModelDebug'.
> A problem occurred starting process 'command '/Users/kaii/Library/Android/sdk/ndk/21.0.6224779/ndk-build''

BUILD FAILED in 21s
Gradle task assembleDebug failed with exit code 1
Exited (sigterm)
```

**Comment by simolus3** [comment]
Thanks for trying it out! Can you run `flutter build apk --verbose` to give more details on why the build fails? Thanks!

If you need an urgent fix, you could try using `moor_flutter` instead of `moor_ffi`. Ideally it should be a [one line change](https://pub.dev/packages/moor_flutter#generating-the-code) in the database constructor. Moor ffi is generally better though, so we should investigate why the build is failing.

**Comment by ykaito21** [comment]
I tried `flutter build apk --verbose` and this is the log
```
[  +28 ms] executing: [/Users/kaii/development/flutter/] git -c log.showSignature=false log -n 1
--pretty=format:%H
[  +61 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 27321ebbad34b0a3fafe99fac037102196d655ff
[        ] executing: [/Users/kaii/development/flutter/] git describe --match v*.*.* --first-parent
--long --tags
[  +50 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[        ] v1.12.13+hotfix.5-0-g27321ebba
[   +9 ms] executing: [/Users/kaii/development/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[   +8 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [/Users/kaii/development/flutter/] git ls-remote --get-url origin
[   +7 ms] Exit code 0 from: git ls-remote --get-url origin
[   +1 ms] https://github.com/flutter/flutter.git
[  +69 ms] executing: [/Users/kaii/development/flutter/] git rev-parse --abbrev-ref HEAD
[  +16 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[  +22 ms] executing: sw_vers -productName
[  +46 ms] Exit code 0 from: sw_vers -productName
[        ] Mac OS X
[        ] executing: sw_vers -productVersion
[  +31 ms] Exit code 0 from: sw_vers -productVersion
[   +1 ms] 10.14.6
[        ] executing: sw_vers -buildVersion
[  +19 ms] Exit code 0 from: sw_vers -buildVersion
[        ] 18G95
[  +37 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[   +5 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +180 ms] Found plugin moor_ffi at
/Users/kaii/development/flutter/.pub-cache/git/moor-10b4df22e6aeecdcc28f32c411b4d54de82af6b2/moor_ff
i/
[  +20 ms] Found plugin path_provider at
/Users/kaii/development/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-1.5.1/
[ +102 ms] Found plugin moor_ffi at
/Users/kaii/development/flutter/.pub-cache/git/moor-10b4df22e6aeecdcc28f32c411b4d54de82af6b2/moor_ff
i/
[   +5 ms] Found plugin path_provider at
/Users/kaii/development/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-1.5.1/
[  +74 ms] Generating
/Users/kaii/project/flutter/whatto/android/app/src/main/java/io/flutter/plugins/Gene
ratedPluginRegistrant.java
[  +53 ms] executing: /usr/bin/xcodebuild -version
[ +104 ms] Exit code 0 from: /usr/bin/xcodebuild -version
[        ] Xcode 10.3
           Build version 10G8
[  +27 ms] You are building a fat APK that includes binaries for android-arm, android-arm64, android-x64.
[        ] If you are deploying the app to the Play Store, it's recommended to use app bundles or split
the APK to
           reduce the APK size.
[        ]     To generate an app bundle, run:
[        ]         flutter build appbundle --target-platform android-arm,android-arm64,android-x64
[        ]         Learn more on: https://developer.android.com/guide/app-bundle
[        ]     To split the APKs per ABI, run:
[        ]         flutter build apk --target-platform android-arm,android-arm64,android-x64
--split-per-abi
[        ]         Learn more on:
                   https://developer.android.com/studio/build/configure-apk-splits#configure-abi-split
[  +23 ms] Running Gradle task 'assembleRelease'...
[   +1 ms] gradle.properties already sets `android.enableR8`
[   +4 ms] Using gradle from /Users/kaii/project/flutter/whatto/android/gradlew.
[ +228 ms] executing: /usr/bin/plutil -convert json -o - /Applications/Android
Studio.app/Contents/Info.plist
[  +13 ms] Exit code 0 from: /usr/bin/plutil -convert json -o - /Applications/Android
Studio.app/Contents/Info.plist
[   +1 ms] {"CFBundleName":"Android
Studio","JVMOptions":{"MainClass":"com.intellij.idea.Main","ClassPath":"$APP_PACKAGE\/Contents\/lib\/boot
strap.jar:$APP_PACKAGE\/Contents\/lib\/extensions.jar:$APP_PACKAGE\/Contents\/lib\/util.jar:$APP_PACKAGE\
/Contents\/lib\/jdom.jar:$APP_PACKAGE\/Contents\/lib\/log4j.jar:$APP_PACKAGE\/Contents\/lib\/trove4j.jar:
$APP_PACKAGE\/Contents\/lib\/jna.jar","JVMVersion":"1.8*,1.8+","Properties":{"idea.home.path":"$APP_PACKA
GE\/Contents","idea.executable":"studio","idea.platform.prefix":"AndroidStudio","idea.paths.selector":"An
droidStudio3.5"},"WorkingDirectory":"$APP_PACKAGE\/Contents\/bin"},"LSArchitecturePriority":["x86_64"],"C
FBundleVersion":"AI-191.8026.42.35.5977832","CFBundleDevelopmentRegion":"English","CFBundleDocumentTypes"
:[{"CFBundleTypeExtensions":["ipr"],"CFBundleTypeName":"Android Studio Project
File","CFBundleTypeIconFile":"studio.icns","CFBundleTypeRole":"Editor"},{"CFBundleTypeExtensions":["*"],"
CFBundleTypeOSTypes":["****"],"LSTypeIsPackage":false,"CFBundleTypeName":"All
documents","CFBundleTypeRole":"Editor"}],"NSSupportsAutomaticGraphicsSwitching":true,"CFBundlePackageType
":"APPL","CFBundleIconFile":"studio.icns","NSHighResolutionCapable":true,"CFBundleShortVersionString":"3.
5","CFBundleInfoDictionaryVersion":"6.0","CFBundleExecutable":"studio","LSRequiresNativeExecution":"YES",
"CFBundleURLTypes":[{"CFBundleURLName":"Stacktrace","CFBundleURLSchemes":["idea"],"CFBundleTypeRole":"Edi
tor"}],"CFBundleIdentifier":"com.google.android.studio","LSApplicationCategoryType":"public.app-category.
developer-tools","CFBundleSignature":"????","LSMinimumSystemVersion":"10.8","CFBundleGetInfoString":"Andr
oid Studio 3.5, build AI-191.8026.42.35.5977832. Copyright JetBrains s.r.o., (c) 2000-2019"}
[   +6 ms] executing: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java -version
[ +108 ms] Exit code 0 from: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
-version
[        ] openjdk version "1.8.0_202-release"
           OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
           OpenJDK 64-Bit Server VM (build 25.202-b49-5587405, mixed mode)
[   +2 ms] executing: [/Users/kaii/project/flutter/whatto/android/]
/Users/kaii/project/flutter/whatto/android/gradlew -Pverbose=true
-Ptarget=lib/main.dart -Ptrack-widget-creation=true -Pshrink=true
-Ptarget-platform=android-arm,android-arm64,android-x64 assembleRelease
[+23255 ms] > Task :app:compileFlutterBuildRelease
[        ] [  +26 ms] executing: [/Users/kaii/development/flutter/] git -c log.showSignature=false
log -n 1 --pretty=format:%H
[        ] [  +38 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] [        ] 27321ebbad34b0a3fafe99fac037102196d655ff
[        ] [        ] executing: [/Users/kaii/development/flutter/] git describe --match v*.*.*
--first-parent --long --tags
[        ] [  +51 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[        ] [        ] v1.12.13+hotfix.5-0-g27321ebba
[        ] [  +10 ms] executing: [/Users/kaii/development/flutter/] git rev-parse --abbrev-ref
--symbolic @{u}
[        ] [   +6 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] [        ] origin/stable
[        ] [        ] executing: [/Users/kaii/development/flutter/] git ls-remote --get-url origin
[        ] [   +5 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] [        ] https://github.com/flutter/flutter.git
[   +2 ms] [  +55 ms] executing: [/Users/kaii/development/flutter/] git rev-parse --abbrev-ref HEAD
[        ] [   +7 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] [        ] stable
[        ] [  +25 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping
update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping
update.
[   +4 ms] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [ +128 ms] Initializing file store
[   +6 ms] [  +65 ms] kernel_snapshot: Starting due to {}
[   +1 ms] [  +15 ms] /Users/kaii/development/flutter/bin/cache/dart-sdk/bin/dart
/Users/kaii/development/flutter/bin/cache/artifacts/engine/darwin-x64/frontend_server.dart.snapshot
--sdk-root
/Users/kaii/development/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk_product/
--target=flutter -Ddart.developer.causal_async_stacks=true -Ddart.vm.profile=false -Ddart.vm.product=true
--bytecode-options=source-positions --aot --tfa --packages
/Users/kaii/project/flutter/whatto/.packages --output-dill
/Users/kaii/project/flutter/whatto/.dart_tool/flutter_build/dc1bee56cd9a3f279d3401ff
1942d0be/app.dill --depfile
/Users/kaii/project/flutter/whatto/.dart_tool/flutter_build/dc1bee56cd9a3f279d3401ff
1942d0be/kernel_snapshot.d package:whattoeat_app/main.dart
[+21260 ms] [+22183 ms] kernel_snapshot: Complete
[+1503 ms] [+1512 ms] aot_android_asset_bundle: Starting due to {}
[ +402 ms] [ +341 ms] aot_android_asset_bundle: Complete
[ +496 ms] [ +524 ms] android_aot_release_android-arm: Starting due to {InvalidatedReason.inputChanged}
[        ] [   +7 ms] executing:
/Users/kaii/development/flutter/bin/cache/artifacts/engine/android-arm-release/darwin-x64/gen_snapsh
ot --causal_async_stacks --deterministic --snapshot_kind=app-aot-elf
--elf=/Users/kaii/project/flutter/whatto/.dart_tool/flutter_build/dc1bee56cd9a3f279d
3401ff1942d0be/armeabi-v7a/app.so --strip --no-sim-use-hardfp --no-use-integer-division
/Users/kaii/project/flutter/whatto/.dart_tool/flutter_build/dc1bee56cd9a3f279d3401ff
1942d0be/app.dill
[ +101 ms] [ +118 ms] android_aot_release_android-arm64: Starting due to {InvalidatedReason.inputChanged}
[        ] [   +3 ms] executing:
/Users/kaii/development/flutter/bin/cache/artifacts/engine/android-arm64-release/darwin-x64/gen_snap
shot --causal_async_stacks --deterministic --snapshot_kind=app-aot-elf
--elf=/Users/kaii/project/flutter/whatto/.dart_tool/flutter_build/dc1bee56cd9a3f279d
3401ff1942d0be/arm64-v8a/app.so --strip
/Users/kaii/project/flutter/whatto/.dart_tool/flutter_build/dc1bee56cd9a3f279d3401ff
1942d0be/app.dill
[  +99 ms] [  +77 ms] android_aot_release_android-x64: Starting due to {InvalidatedReason.inputChanged}
[        ] [   +1 ms] executing:
/Users/kaii/development/flutter/bin/cache/artifacts/engine/android-x64-release/darwin-x64/gen_snapsh
ot --causal_async_stacks --deterministic --snapshot_kind=app-aot-elf
--elf=/Users/kaii/project/flutter/whatto/.dart_tool/flutter_build/dc1bee56cd9a3f279d
3401ff1942d0be/x86_64/app.so --strip
/Users/kaii/project/flutter/whatto/.dart_tool/flutter_build/dc1bee56cd9a3f279d3401ff
1942d0be/app.dill
[+16197 ms] [+16185 ms] android_aot_release_android-arm64: Complete
[ +100 ms] [  +90 ms] android_aot_bundle_release_android-arm64: Starting due to
{InvalidatedReason.inputChanged}
[        ] [   +9 ms] android_aot_bundle_release_android-arm64: Complete
[ +200 ms] [ +182 ms] android_aot_release_android-x64: Complete
[ +100 ms] [  +81 ms] android_aot_bundle_release_android-x64: Starting due to
{InvalidatedReason.inputChanged}
[        ] [   +8 ms] android_aot_bundle_release_android-x64: Complete
[  +98 ms] [  +82 ms] android_aot_release_android-arm: Complete
[        ] [  +70 ms] android_aot_bundle_release_android-arm: Starting due to
{InvalidatedReason.inputChanged}
[        ] [   +8 ms] android_aot_bundle_release_android-arm: Complete
[  +98 ms] [  +83 ms] _composite: Starting due to {}
[        ] [        ] _composite: Complete
[        ] [        ] Persisting file store
[        ] [   +7 ms] Done persisting file store
[        ] [   +5 ms] build succeeded.
[        ] [  +11 ms] "flutter assemble" took 41,843ms.
[+1308 ms] > Task :app:packLibsflutterBuildRelease
[        ] > Task :app:preBuild UP-TO-DATE
[        ] > Task :app:extractProguardFiles
[        ] > Task :app:preReleaseBuild
[  +92 ms] > Task :moor_ffi:extractSqlite UP-TO-DATE
[        ] > Task :moor_ffi:preBuild UP-TO-DATE
[        ] > Task :moor_ffi:preReleaseBuild UP-TO-DATE
[        ] > Task :moor_ffi:compileReleaseAidl NO-SOURCE
[        ] > Task :path_provider:preBuild UP-TO-DATE
[        ] > Task :path_provider:preReleaseBuild UP-TO-DATE
[        ] > Task :path_provider:compileReleaseAidl NO-SOURCE
[        ] > Task :app:compileReleaseAidl NO-SOURCE
[  +96 ms] > Task :moor_ffi:packageReleaseRenderscript NO-SOURCE
[        ] > Task :path_provider:packageReleaseRenderscript NO-SOURCE
[        ] > Task :app:compileReleaseRenderscript NO-SOURCE
[        ] > Task :app:checkReleaseManifest
[        ] > Task :app:generateReleaseBuildConfig
[        ] > Task :app:cleanMergeReleaseAssets UP-TO-DATE
[        ] > Task :app:mergeReleaseShaders
[        ] > Task :app:compileReleaseShaders
[        ] > Task :app:generateReleaseAssets
[        ] > Task :moor_ffi:mergeReleaseShaders
[        ] > Task :moor_ffi:compileReleaseShaders
[        ] > Task :moor_ffi:generateReleaseAssets
[        ] > Task :moor_ffi:packageReleaseAssets
[        ] > Task :path_provider:mergeReleaseShaders
[        ] > Task :path_provider:compileReleaseShaders
[        ] > Task :path_provider:generateReleaseAssets
[        ] > Task :path_provider:packageReleaseAssets
[        ] > Task :app:mergeReleaseAssets
[  +97 ms] > Task :app:copyFlutterAssetsRelease
[        ] > Task :app:mainApkListPersistenceRelease
[        ] > Task :app:generateReleaseResValues
[        ] > Task :app:generateReleaseResources
[        ] > Task :moor_ffi:compileReleaseRenderscript NO-SOURCE
[        ] > Task :moor_ffi:generateReleaseResValues
[        ] > Task :moor_ffi:generateReleaseResources
[        ] > Task :moor_ffi:packageReleaseResources
[        ] > Task :path_provider:generateReleaseResValues
[        ] > Task :path_provider:compileReleaseRenderscript NO-SOURCE
[        ] > Task :path_provider:generateReleaseResources
[        ] > Task :path_provider:packageReleaseResources
[ +698 ms] > Task :app:createReleaseCompatibleScreenManifests
[        ] > Task :moor_ffi:checkReleaseManifest
[        ] > Task :path_provider:checkReleaseManifest
[  +95 ms] > Task :moor_ffi:processReleaseManifest
[        ] > Task :path_provider:processReleaseManifest
[        ] > Task :app:processReleaseManifest
[        ] > Task :moor_ffi:parseReleaseLibraryResources
[ +101 ms] > Task :path_provider:parseReleaseLibraryResources
[        ] > Task :moor_ffi:generateReleaseBuildConfig
[        ] > Task :moor_ffi:generateReleaseRFile
[ +198 ms] > Task :app:mergeReleaseResources
[        ] > Task :path_provider:generateReleaseRFile
[ +698 ms] > Task :app:processReleaseResources
[        ] > Task :moor_ffi:javaPreCompileRelease
[ +298 ms] > Task :moor_ffi:compileReleaseJavaWithJavac
[        ] > Task :path_provider:generateReleaseBuildConfig
[        ] > Task :moor_ffi:bundleLibCompileRelease
[        ] > Task :path_provider:javaPreCompileRelease
[ +100 ms] > Task :path_provider:compileReleaseJavaWithJavac
[        ] > Task :path_provider:bundleLibCompileRelease
[  +15 ms]
[+1891 ms] > Task :app:compileReleaseKotlin
[        ] > Task :app:javaPreCompileRelease
[ +986 ms] > Task :app:compileReleaseJavaWithJavac
[        ] > Task :app:compileReleaseSources
[        ] > Task :app:prepareLintJar
[        ] > Task :moor_ffi:prepareLintJarForPublish
[        ] > Task :app:validateSigningRelease
[        ] > Task :app:signingConfigWriterRelease
[        ] > Task :path_provider:prepareLintJarForPublish
[+1999 ms] > Task :app:lintVitalRelease
[        ] Calling mockable JAR artifact transform to create file:
/Users/kaii/.gradle/caches/transforms-2/files-2.1/98ba7a917f2aa7f2f92d2793902a7b10/android.jar with
input /Users/kaii/Library/Android/sdk/platforms/android-28/android.jar
[+10703 ms] > Task :app:mergeReleaseJniLibFolders
[        ] > Task :moor_ffi:processReleaseJavaRes NO-SOURCE
[        ] > Task :moor_ffi:generateJsonModelRelease FAILED
[        ] > Task :moor_ffi:bundleLibResRelease
[ +197 ms] FAILURE: Build failed with an exception.
[        ] * What went wrong:
[        ] Execution failed for task ':moor_ffi:generateJsonModelRelease'.
[        ] > A problem occurred starting process 'command
'/Users/kaii/Library/Android/sdk/ndk/21.0.6113669/ndk-build''
[        ] * Try:
[        ] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get
more log output. Run with --scan to get full insights.
[        ] * Get more help at https://help.gradle.org
[        ] BUILD FAILED in 1m 23s
[        ] Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
[   +3 ms] Use '--warning-mode all' to show the individual deprecation warnings.
[        ] See
https://docs.gradle.org/5.6.2/userguide/command_line_interface.html#sec:command_line_warnings
[        ] 55 actionable tasks: 53 executed, 2 up-to-date
[ +431 ms] Running Gradle task 'assembleRelease'... (completed in 84.3s)
[   +7 ms] "flutter apk" took 85,147ms.
Gradle task assembleRelease failed with exit code 1

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      buildGradleApp (package:flutter_tools/src/android/gradle.dart:387:7)
#2      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:73:64)
#3      _rootRunUnary (dart:async/zone.dart:1134:38)
#4      _CustomZone.runUnary (dart:async/zone.dart:1031:19)
#5      _FutureListener.handleValue (dart:async/future_impl.dart:139:18)
#6      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:680:45)
#7      Future._propagateToListeners (dart:async/future_impl.dart:709:32)
#8      Future._completeWithValue (dart:async/future_impl.dart:524:5)
#9      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:32:15)
#10     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:290:13)
#11     _DefaultProcessUtils.stream (package:flutter_tools/src/base/process.dart)
#12     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:73:64)
#13     _rootRunUnary (dart:async/zone.dart:1134:38)
#14     _CustomZone.runUnary (dart:async/zone.dart:1031:19)
#15     _FutureListener.handleValue (dart:async/future_impl.dart:139:18)
#16     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:680:45)
#17     Future._propagateToListeners (dart:async/future_impl.dart:709:32)
#18     Future._completeWithValue (dart:async/future_impl.dart:524:5)
#19     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:554:7)
#20     _rootRun (dart:async/zone.dart:1126:13)
#21     _CustomZone.run (dart:async/zone.dart:1023:19)
#22     _CustomZone.runGuarded (dart:async/zone.dart:925:7)
#23     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:965:23)
#24     _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
#25     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)
#26     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#27     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:175:5)
```

Thanks for your help!

**Comment by ykaito21** [comment]
It was actually not problem with moor package. I tried `moor_flutter` but it also failed to build.
From new log, I reached to [Android App Bundle does not build with Gradle 3.5.0 #38934](https://github.com/flutter/flutter/issues/38934).

I changed `classpath 'com.android.tools.build:gradle:3.5.0'` to `classpath 'com.android.tools.build:gradle:3.4.2'`, and it works fine now with both `moor_fill` and `moor_flutter`. 

I don't know how to match proper gradle version, but anyway thank you so much!
